### PR TITLE
Simplify deduction rule

### PR DESCRIPTION
**Pull request by ngeiswei**
PLN rules coders, please have a look at this, you don't need to duplicate over and over the rule code, use a rule generator instead.

### CODE REVIEW

File: opencog/reasoning/pln/rules/deduction.scm [hunk 1]
```diff
@@ -1,102 +0,0 @@
...
-            (InheritanceLink
-                (VariableNode "$A")
-                (VariableNode "$B")
-            )
-            (InheritanceLink
-                (VariableNode "$B")
-                (VariableNode "$C")
-            )
-            ; To avoid matching (Inheritance A B) and (Inheritance B A)
-            (NotLink
```
Comment (williampma) [inline opencog/reasoning/pln/rules/deduction.scm]: This part is missing on the new deduction rule.

**Comment by ngeiswei** [comment]
@linas maybe you can close this PR, I'm gonna do a grand PLN rule cleaning and I'll clean those variables in the process.

In meantime it would be good to find out while PLNRulesUTest fails on your system...

**Comment by linas** [comment]
?? which PR?

**Comment by ngeiswei** [comment]
Never mind, I'm also doing to many things in parallel ;-)

File: opencog/reasoning/pln/rules/deduction.scm (hunk 1 above)
Comment (ngeiswei) [inline opencog/reasoning/pln/rules/deduction.scm]: Good catch, thanks!

Comment (ngeiswei) [inline opencog/reasoning/pln/rules/deduction.scm]: Hmm, and PLNRuleUTest didn't catch it, I understand why, it doesn't produce invalid outputs, just useless...

**Pull request merged**